osu.Game.Rulesets.Osu/UI/Cursor/CursorRippleVisualiser.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Pooling;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+using osu.Game.Rulesets.Osu.Configuration;
+using osu.Game.Rulesets.Osu.Skinning.Default;
+using osu.Game.Skinning;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.UI.Cursor
+{
+    public partial class CursorRippleVisualiser : CompositeDrawable, IKeyBindingHandler<OsuAction>
+    {
+        private readonly Bindable<bool> showRipples = new Bindable<bool>(true);
+
+        private readonly DrawablePool<CursorRipple> ripplePool = new DrawablePool<CursorRipple>(20);
+
+        public CursorRippleVisualiser()
+        {
+            RelativeSizeAxes = Axes.Both;
+        }
+
+        [BackgroundDependencyLoader(true)]
+        private void load(OsuRulesetConfigManager? rulesetConfig)
+        {
+            rulesetConfig?.BindWith(OsuRulesetSetting.ShowCursorRipples, showRipples);
+        }
+
+        public bool OnPressed(KeyBindingPressEvent<OsuAction> e)
+        {
+            if (showRipples.Value)
+            {
+                AddInternal(ripplePool.Get(r =>
+                {
+                    r.Position = e.MousePosition;
+                }));
+            }
+
+            return false;
+        }
+
+        public void OnReleased(KeyBindingReleaseEvent<OsuAction> e)
+        {
+        }
+
+        private partial class CursorRipple : PoolableDrawable
+        {
+            [BackgroundDependencyLoader]
+            private void load()
+            {
+                AutoSizeAxes = Axes.Both;
+                Origin = Anchor.Centre;
+
+                InternalChildren = new Drawable[]
+                {
+                    new SkinnableDrawable(new OsuSkinComponentLookup(OsuSkinComponents.CursorRipple), _ => new DefaultCursorRipple())
+                    {
+                        Blending = BlendingParameters.Additive,
+                    }
+                };
+            }
+
+            protected override void PrepareForUse()
+            {
+                base.PrepareForUse();
+
+                ClearTransforms(true);
+
+                this.ScaleTo(0.05f)
+                    .ScaleTo(1, 700, Easing.Out)
```

Comment:
Looking at stable source, it does a `0.5 -> 0.05` transform - so if we're halving sizes to get rid of the legacy sprite scaling logic, then shouldn't this go from `0.1f` to `1`?

Reply:
Good catch.